velox/substrait/SubstraitToVeloxPlan.cpp

```diff
@@ -203,15 +204,30 @@ std::shared_ptr<const core::PlanNode> SubstraitVeloxPlanConverter::toVeloxPlan(
 
 std::shared_ptr<const core::PlanNode> SubstraitVeloxPlanConverter::toVeloxPlan(
     const ::substrait::FilterRel& sFilter) {
-  // TODO: Currently Filter is skipped because Filter is Pushdowned to
-  // TableScan.
   std::shared_ptr<const core::PlanNode> childNode;
   if (sFilter.has_input()) {
     childNode = toVeloxPlan(sFilter.input());
   } else {
     VELOX_FAIL("Child Rel is expected in FilterRel.");
   }
-  return childNode;
+  // This is for the case when the filter is not be pushdowned by TableScan.
+  // TODO remove the bool condition after the sub.json remove the filter node
+  bool filterPushdownEnabled = true;
+  if (filterPushdownEnabled) {
+    return childNode;
+  } else {
+    RowTypePtr vPreNodeOutPut = childNode->outputType();
+    if (!sFilter.has_condition()) {
```

Comment:
How is this possible? What does it mean to have a filter without the condition?

Reply:
remove this.